### PR TITLE
[EFR32] Implement persistence for operational keys

### PR DESCRIPTION
**Pull request by rcasallas-silabs**
#### Problem
Missing support for EFR32 vault-protected private keys

#### Change overview
* Fix flash start address on xG24 for running examples
* Add a way to use a platform-defined crypto backend
* Add a PSA Crypto backend for EFR32
* Finetune PSA Crypto configuration
* Now that we have HMAC through PSA, we don't need PKCS5 anymore
* Add implementation of opaque keys and opaque operational key store
* Set correct SHA context size on devices with HW acceleration

#### Testing
* TODO

### CODE REVIEW

File: src/platform/EFR32/CHIPCryptoPALPsaEfr32.cpp [hunk 1]
```diff
@@ -0,0 +1,1939 @@
+/*
```
Comment (Damian-Nordic) [inline src/platform/EFR32/CHIPCryptoPALPsaEfr32.cpp:1]: Does it use any Sillicon Labs APIs or only mbedTLS/PSA crypto? I wonder if we could move it to `src/crypto` and reuse in other platforms easily as we were also planning working on that.

Comment (stevew817) [reply src/platform/EFR32/CHIPCryptoPALPsaEfr32.cpp:1]: It takes shortcuts by calling into the driver wrapper layer (to bypass having to import/delete a key in the PSA key database for every single operation), which is not declared a stable API. That's why IMO this implementation doesn't belong in `src/crypto` (and neither does the SE050 implementation, but that's another discussion).

You could easily build on it and make a generic mbedTLS/PSA implementation of the Matter Crypto HAL (just change out the `psa_driver_wrapper_xxx()` calls for their top-level equivalents and wrap them in a `psa_import_key()`/`psa_destroy_key()` pair), but it'd suffer from performance issues making it not worth it until Matter moves to abstract keys for the symmetric operations as well.

That is, unless you're trying to build for something like TF-M. Then I guess eating the overhead is your only option besides duplicating the symmetric crypto algorithm implementations in the non-secure world.

Comment (stevew817) [reply src/platform/EFR32/CHIPCryptoPALPsaEfr32.cpp:1]: A PSA API-compliant implementation has been prototyped here: https://github.com/stevew817/connectedhomeip/blob/5b1ad7f945ee7ffe67ee7633f79f7cdce20defa0/src/crypto/CHIPCryptoPALmbedTLS.cpp

Comment (Damian-Nordic) [reply src/platform/EFR32/CHIPCryptoPALPsaEfr32.cpp:1]: I see, thanks very much for the explanation. Yeah, I think we need to introduce a better abstraction for keys in Matter to be prepared for TF-M. :)

**Pull request closed without merging**